fapolicy_analyzer/ui/rules/rules_admin_page.py

```diff
@@ -121,11 +121,17 @@ def __load_rules(self):
     def __rules_unvalidated(self) -> bool:
         return not self.__rules_validated
 
+    def get_is_dirty(self):
+        return self.__rules_dirty()
+
     def __rules_dirty(self) -> bool:
-        return (
+        is_dirty = (
             bool(self.__modified_rules_text)
             and self.__modified_rules_text != self.__rules_text
         )
+        text = "*Rules Editor" if is_dirty else "Rules Editor"
+        self.get_object("editorView").set_text(text)
```

Comment:
A better place for this code is in the `on_text_view_rules_changed` event handler. You just need to do your own comparison there between modified rules and original rules.